src/send/mod.rs

```diff
@@ -64,24 +56,6 @@ impl FromData for Email {
     }
 }
 
-fn validate(email: &Email) -> bool {
-    if let Some(ref cc) = email.cc {
-        for address in cc {
-            if !validate::email_address(&address) {
-                return false;
-            }
-        }
-    }
```

Comment:
Again, all this ugly stuff can go now because the validation is already encapsulated inside `EmailAddress`.